Playdown.swift

```diff
@@ -130,61 +126,110 @@ extension String {
         }
     }
     
-    func subrangesMatchingPattern(let pattern: String, let options: NSRegularExpressionOptions) throws -> [NSRange] {
+    func subrangesMatchingPattern(_ pattern: String, options: NSRegularExpression.Options) throws -> [NSRange] {
         let range = NSMakeRange(0, (self as NSString).length)
         let regex = try NSRegularExpression(pattern: pattern, options: options)
-        let matches = regex.matchesInString(self, options: [], range: range)
-        return matches.map { return $0.rangeAtIndex(0) }
+        let matches = regex.matches(in: self, options: [], range: range)
+        return matches.map { return $0.rangeAt(0) }
+    }
+    
+   func strip(from char:Character) -> String {
+        let c = self.characters
+        if let ix = c.index(of: char) {
+            return String(c.prefix(upTo: ix))
+        }
+        return ""
     }
 }
 
 struct Playdown {
-    let streamReader: StreamReader!
+    
+    var streamReaders: [(reader: StreamReader, name: String)] = []
     let SingleLineTextBeginningPattern = "^//:"
     let MultilineTextBeginningPattern = "/\\*:"
     let MultilineTextEndingPattern = "\\*/"
     let MarkdownCodeStartDelimiter = "```swift"
     let MarkdownCodeEndDelimiter = "```\n"
 
     enum LineType {
-        case SingleLineText, MultilineText, SwiftCode
+        case singleLineText, multilineText, swiftCode
     }
     
     init(filename: String) {
-        streamReader = StreamReader(path: filename)
+        
+        let fileManager = FileManager.default
+        let path =  fileManager.currentDirectoryPath + "/" + filename + "/Pages/"
```

Comment:
I originally tested to make sure it worked with one page using the Swift Style Guide - which only has one page. I now realize that brand new Playgrounds don't have the Pages directory :/ because technically new Playgrounds don't even have one page; it is just the playground with a Contents.swift file of it. Would you like a markdown to be created based off of the Playground name in that case?